pandas/tests/io/parser/usecols.py

```diff
@@ -481,7 +481,7 @@ def test_raise_on_usecols_names_mismatch(self):
         data = 'a,b,c,d\n1,2,3,4\n5,6,7,8'
 
         if self.engine == 'c':
-            msg = 'Usecols do not match names'
+            msg = "do not match columns, columns expected but not found"
```

Comment:
Agree on the test, it'll require a minor reformat of the structure of the test - if that's not an issue happy to go ahead (unless wildcarding the regex for the arguments is sufficient).

Will see if it's also an issue on the Python engine and if so will implement a fix!